chore: cache reroute results (#13548)
sveltejs/kit: documentation/docs/30-advanced/20-hooks.md

Since version 2.18, the `reroute` hook can be asynchronous, allowing it to (for example) fetch data from your backend to decide where to reroute to. Use this carefully and make sure it's fast, as it will delay navigation otherwise.

> > [!NOTE] `reroute` is considered a pure, idempotent function. That means, it should always return the same output for the same input and not have side effects. Under these assumptions, SvelteKit caches the result of `reroute` on the client so it is only  called once per unique URL.

Reviewer: ```suggestion
> [!NOTE] `reroute` is considered a pure, idempotent function. As such, it must always return the same output for the same input and not have side effects. Under these assumptions, SvelteKit caches the result of `reroute` on the client so it is only called once per unique URL.
```
